c_src/dpiStmt_nif.c

```diff
@@ -297,6 +297,105 @@ DPI_NIF_FUN(stmt_close)
     return ATOM_OK;
 }
 
+DPI_NIF_FUN(stmt_getInfo)
+{
+    CHECK_ARGCOUNT(1);
+
+    dpiStmt_res *stmtRes = NULL;
+    dpiStmtInfo info;
+
+    if (!enif_get_resource(env, argv[0], dpiStmt_type, (void **)&stmtRes))
+        BADARG_EXCEPTION(0, "resource statement");
+
+    RAISE_EXCEPTION_ON_DPI_ERROR(
+        stmtRes->context,
+        dpiStmt_getInfo(stmtRes->stmt, &info),
+        stmtRes);
+
+    ERL_NIF_TERM map = enif_make_new_map(env);
+
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "isDDL"),
+        info.isDDL ? ATOM_TRUE : ATOM_FALSE,
+        &map);
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "isDML"),
+        info.isDML ? ATOM_TRUE : ATOM_FALSE,
+        &map);
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "isPLSQL"),
+        info.isPLSQL ? ATOM_TRUE : ATOM_FALSE,
+        &map);
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "isQuery"),
+        info.isQuery ? ATOM_TRUE : ATOM_FALSE,
+        &map);
+    enif_make_map_put(
+        env, map, enif_make_atom(env, "isReturning"),
+        info.isReturning ? ATOM_TRUE : ATOM_FALSE,
+        &map);
```

Comment:
error handle [`enif_make_map_put`](http://erlang.org/doc/man/erl_nif.html#enif_make_map_put)

Reply:
map is created locally, thus it is assured that it is a valid map (and `enif_make_map_put` cannot fail otherwise)